### DISCUSSION TITLE
Swiftlint Type Violation

### DISCUSSION THREAD

**Discussion by sherryycxie** [CS342/.github#20] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

I'm consistently getting this error message from SwiftLint: "Type Contents Order Violation: An 'instance_property' should not be placed amongst the type content(s) 'type_property'. (type_contents_order)," which I believe is related to these lines of code: 
<img width="991" alt="Screenshot 2023-01-24 at 9 29 36 PM" src="https://user-images.githubusercontent.com/85938088/214487340-0d449dc8-02df-4456-9684-b7bb156270f8.png">
but i'm not quite sure how to resolve this problem.

### Reproduction

Going online to look for this error message, and changing different variable types

### Expected behavior

This should not produce a code check error

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment, marked as answer]
Hey @sherryycxie, the issue here is that SwiftLint wants your type properties (i.e. static properties) to be declared first in your struct, then your instance properties:

```swift
struct Page: Identifiable, Equatable {
      static var samplePage = ...
      static var samplePages = ...

      let id = UUID()
      var name: String
      var description: String 
      var imageUrl: String
      var tag: Int
}
```

You can see an example of the full order of subtypes, properties, methods, etc. here: https://realm.github.io/SwiftLint/type_contents_order.html

**Reply by sherryycxie** [reply]
Thanks Vishnu, that's very helpful! After making this change, I don't have the error message anymore.

**Reply by vishnuravi** [reply]
Great @sherryycxie, glad to help!
